dlrover/python/master/diagnosis/diagnosis_data_manager.py

```diff
@@ -0,0 +1,52 @@
+# Copyright 2024 The DLRover Authors. All rights reserved.
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import Dict, List
+
+from dlrover.python.diagnosis.common.diagnosis_data import DiagnosisData
+from dlrover.python.util.time_util import has_expired
+
+
+class DiagnosisDataManager:
+    def __init__(self, expire_time_period):
+        """
+        :param expire_time_period: data expire time period in seconds
+        """
+        self.diagnosis_data: Dict[str, List[DiagnosisData]] = {}
```

Comment:
Need limited size collections here instead of ```List```.

Reply:
Here we remove expired data periodically and it is easier to remove expired data in list.

Reply:
Size limitedSize limits are necessary because you cannot predict the number of reports submitted within a unit of time. e.g. 5mb data report per unit of time(by thousands of nodes)